dds_web/api/project.py

```diff
@@ -188,6 +188,107 @@ def post(self):
 
         return {"message": return_message}
 
+    @auth.login_required(role=["Unit Admin", "Unit Personnel"])
+    @logging_bind_request
+    @json_required
+    @handle_validation_errors
+    def patch(self):
+        """Partially update a the project status"""
+        # Get project ID, project and verify access
+        project_id = dds_web.utils.get_required_item(obj=flask.request.args, req="project")
+        project = dds_web.utils.collect_project(project_id=project_id)
+        dds_web.utils.verify_project_access(project=project)
+
+        # get atributes
+        json_input = flask.request.get_json(silent=True)  # Already checked by json_required
+
+        # false by default - operation must be confirmed by the user
+        confirmed_operation = json_input.get("confirmed", False)
+        if not isinstance(confirmed_operation, bool):
+            raise DDSArgumentError(message="`confirmed` is a boolean value: True or False.")
+        if not confirmed_operation:
+            warning_message = "Operation must be confirmed before proceding."
+            return {"warning": warning_message}
+
+        # Cannot change project status if project is busy
+        if project.busy:
+            raise ProjectBusyError(
+                message=(
+                    f"The deadline for the project '{project_id}' is already in the process of being changed. "
+                    "Please try again later. \n\nIf you know the project is not busy, contact support."
+                )
+            )
+
+        self.set_busy(project=project, busy=True)
+
+        try:
+            extend_deadline = json_input.get("extend_deadline", False)  # False by default
```

Comment:
This is a good check to have if the endpoint had more options, but since we're only allowing extending the deadline for now, it's better to leave this out and only add it if we need it in the future, to keep the code as simple as possible.

Reply:
Oka, I will make then the check with:

```
            new_deadline_in = json_input.get(
                "new_deadline_in", None
            )  # if not provided is None -> deadlie is not updated
```

Then, we just need to add the atributes to update in the future if the function is expanded